processing/src/main/java/io/druid/segment/serde/DoubleGenericColumnPartSerdeV2.java

```diff
@@ -0,0 +1,188 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.serde;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Suppliers;
+import com.google.common.primitives.Ints;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.java.util.common.IAE;
+import io.druid.java.util.common.io.smoosh.FileSmoosher;
+import io.druid.segment.DoubleColumnSerializer;
+import io.druid.segment.column.ValueType;
+import io.druid.segment.data.BitmapSerde;
+import io.druid.segment.data.BitmapSerdeFactory;
+import io.druid.segment.data.ByteBufferWriter;
+import io.druid.segment.data.CompressedDoublesIndexedSupplier;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.WritableByteChannel;
+
+public class DoubleGenericColumnPartSerdeV2 implements ColumnPartSerde
+{
+  static final byte VERSION_ONE = 0x1;
+  private final ByteOrder byteOrder;
+  private Serializer serialize;
+  private final BitmapSerdeFactory bitmapSerdeFactory;
+
+  @JsonCreator
+  public static DoubleGenericColumnPartSerdeV2 getDoubleGenericColumnPartSerde(
+      @JsonProperty("byteOrder") ByteOrder byteOrder,
+      @Nullable @JsonProperty("bitmapSerdeFactory") BitmapSerdeFactory bitmapSerdeFactory
+  )
+  {
+    return new DoubleGenericColumnPartSerdeV2(byteOrder,
+                                              bitmapSerdeFactory != null
+                                              ? bitmapSerdeFactory
+                                              : new BitmapSerde.LegacyBitmapSerdeFactory(), null
+    );
+  }
+
+  @JsonProperty
+  public ByteOrder getByteOrder()
+  {
+    return byteOrder;
+  }
+
+
+  public DoubleGenericColumnPartSerdeV2(
+      ByteOrder byteOrder,
+      BitmapSerdeFactory bitmapSerdeFactory,
+      Serializer serialize
+  )
+  {
+    this.byteOrder = byteOrder;
+    this.bitmapSerdeFactory = bitmapSerdeFactory;
+    this.serialize = serialize;
+  }
+
+  @Override
+  public Serializer getSerializer()
+  {
+    return serialize;
+  }
+
+  @Override
+  public Deserializer getDeserializer()
+  {
+    return (buffer, builder, columnConfig) -> {
+      byte versionFromBuffer = buffer.get();
+
+      if (VERSION_ONE == versionFromBuffer) {
+
+        int offset = buffer.getInt();
+        int initialPos = buffer.position();
+        final CompressedDoublesIndexedSupplier column = CompressedDoublesIndexedSupplier.fromByteBuffer(
+            buffer,
+            byteOrder,
+            builder.getFileMapper()
+        );
+
+        buffer.position(initialPos + offset);
+        final ImmutableBitmap bitmap;
+        if (buffer.hasRemaining()) {
+          int bitmapSize = buffer.getInt();
+          bitmap = bitmapSerdeFactory.getObjectStrategy().fromByteBuffer(buffer, bitmapSize);
```

Comment:
is there any reason this is different from the Float one? io.druid.segment.data.ByteBufferSerializer#read ?